craft_parts/lifecycle_manager.py

```diff
@@ -110,21 +119,41 @@ def __init__(
         for name, spec in parts_data.items():
             part_list.append(_build_part(name, spec, project_dirs))
 
+        self._has_overlay = any(p.has_overlay for p in part_list)
+
+        # a base layer is mandatory if overlays are in use
+        if self._has_overlay:
+            if not base_layer_dir:
+                raise ValueError("using overlays and base_layer_dir not specified")
+            if not base_layer_hash:
+                raise ValueError("using overlays and base_layer hash not specified")
+        else:
+            base_layer_dir = None
+
+        if base_layer_hash:
+            layer_hash: Optional[LayerHash] = LayerHash(base_layer_hash)
+        else:
+            layer_hash = None
```

Comment:
Having base_layer_dir and/or layer_hash is ok if NOT using overlays?

Also, if one of base_layer_dir or layer_hash is present, the other must be present too?

Reply:
If not using overlays, the base layer is not mounted and the overlay verification hash will not be computed, so the parameters are ignored.

Reply:
I see, so we're "safe". My question, though, was coming from the point of view of passing this information.

So, if I pass A and not B or C, instead of part getting ignored, shouldn't it raise an error?